### PR TITLE
Download dataset

### PR DESCRIPTION
**Pull request by ElieKadoche**
I created scripts to download raw data (`.csv`) and to create datasets (`.npy`). I created `.gitignore`, `requirements.txt` and I updated the `README.md`. You can read it to understand my work. I created the basic architecture for the project.

Now, our data is in the `dataset/` folder, in the form of numpy arrays. Carefull, when you load `x_train` and `x_test` you need to add `allow_pickle=True` because values are of type string.

```python
x_train = np.load("x_train.npy", allow_pickle=True)
x_test = np.load("x_test.npy", allow_pickle=True)
```

All paths are in the `utils` folder. In the Sentiment140 train dataset, all labels are 0 and 4, but there are none 1, 2 and 3. I found it strange. Please review my code to make sure there is no mistake. Tell me if you do not agree with choices I made.

After that, we will be able, each one of us, to test different implementations. The goal is simple : create a model with the highest accuracy. A model takes as input a string of variable lenght and outputs a sentiment class between 0 and 4.

### CODE REVIEW

**Comment by fmdinh** [comment]
Thanks Elie for the work! In the data loading, I guess it will be better if we can separate the data by Train, Test and Validation!

In fact, we have 160k data for the train and only 498 data for the test. I propose to separate the data by using train_test_split function: therefore 136k data for the train, 24k for the test and 498 for the validation!

```
x_data = csv_train["text"].to_numpy()    
x_valid_data = csv_valid["text"].to_numpy()
y_data = csv_train["polarity"].to_numpy().astype(int)
y_valid_data = csv_valid["polarity"].to_numpy().astype(int)
    
(x_train_data, x_test_data, y_train_data, y_test_data) = train_test_split(x_data, y_data, test_size=0.15, random_state=42)
    
```

**Comment by ElieKadoche** [comment]
This is a good idea, but the reason I did not do it is the following.

We are absolutely sure that the train set and the test set are fixed, for all models. But splitting the training data into a valid and a train set, is a choice to train a model. All models would not necessarily do that.

Therefore, splitting the training data into a valid and a train set should be done into the training script of a specific model. It should not be done globally.

**Comment by ElieKadoche** [comment]
Little note: for the next part of the project, I propose the following architecture.

Simply create a single script per solution at the root project. A solution can save models in `models/name_of_the_solution/` and results in `results/name_of_the_solution/`. Every solution must implements a function to compute accuracy over the test dataset.

**Comment by fmdinh** [comment]
Hi Elie! Thanks for the remark!

I think it will be good if we can agree a ratio of the test and train dataset in our local model - so that our model can be transparent for comparison. I guess we divided 80% for the train and 20% for the validation test?

**Comment by ElieKadoche** [comment]
Yes, good idea!

So 20% of the training set will be our validation set. But since there is no constraint about it, we can do the split as we like. The only important thing, is not to use the test set as a validation set. That is to say, not tune our models thanks to the test set.

File: README.md [hunk 1]
```diff
@@ -1,2 +1,20 @@
 # deep_sentiment140
-Deep learning approaches for sentiment140 dataset.
+
+Deep learning approaches for Sentiment140 dataset.
+
+## Prerequisites
+
+You need to have Python 3.6 (at least) installed on your system. To run this project, commands are the following.
```
Comment (XanX3601) [inline README.md]:
```suggestion
 [Python](https://www.python.org/) 3.6 or higher is required.
 
 All python requirements can be found in file `requirements.txt` and can be installed using
 
 ```shell
 pip install -r requirements.txt
 ```
```

File: create_datasets.py [hunk 2]
```diff
@@ -0,0 +1,40 @@
+"""Script to create Sentiment140 datasets."""
+
+import argparse
+
+import numpy as np
+import pandas as pd
+
+import utils
+
+parser = argparse.ArgumentParser(
```
Comment (XanX3601) [inline create_datasets.py]: The parser is only necessary when runing the script directly therefore, I would put it inside the `if __name__ == '__main__'`condition

File: download_dataset.py [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+import io
+import zipfile
+
+import requests
+
+import utils
+
+DATASET_URL = "http://cs.stanford.edu/people/alecmgo/trainingandtestdata.zip"
+
+parser = argparse.ArgumentParser(
```
Comment (XanX3601) [inline download_dataset.py]: The parser is only necessary when runing the script directly; therefore I would put it in the `if __name__ == '__main__'` condition

File: download_dataset.py [hunk 4]
```diff
@@ -0,0 +1,26 @@
...
+
+if __name__ == "__main__":
+    args = parser.parse_args()
+    print("Downloading Sentiment140 dataset...")
+
+    utils.create_dir(utils.DATASET_FOLDER, delete_old=True)
+
+    r = requests.get(DATASET_URL)
+    z = zipfile.ZipFile(io.BytesIO(r.content))
+    z.extractall(utils.DATASET_FOLDER)
```
Comment (XanX3601) [inline download_dataset.py:25]: I did not check but if the file is heavy, a progress bar would be a nice touch, especially when using low connection

File: requirements.txt [hunk 5]
```diff
@@ -0,0 +1,5 @@
+argparse
```
Comment (XanX3601) [inline requirements.txt]: Required true but a default package of pyton, isn't it ?

File: utils/folders.py [hunk 6]
```diff
@@ -0,0 +1,24 @@
+"""Folders tools."""
+
+import os
+
+DATASET_FOLDER = "dataset/"
+RESULTS_FOLDER = "results/"
+MODELS_FOLDER = "models/"
+
+
+def create_dir(directory_path, delete_old=False):
```
Comment (XanX3601) [inline utils/folders.py:10]: I like that

File: utils/datasets.py [hunk 7]
```diff
@@ -0,0 +1,11 @@
...
+
+TRAIN_CSV_PATH = "{}training.1600000.processed.noemoticon.csv".format(
+    DATASET_FOLDER)
+TEST_CSV_PATH = "{}testdata.manual.2009.06.14.csv".format(DATASET_FOLDER)
+
+X_TRAIN_PATH = "{}x_train.npy".format(DATASET_FOLDER)
+Y_TRAIN_PATH = "{}y_train.npy".format(DATASET_FOLDER)
+
+X_TEST_PATH = "{}x_test.npy".format(DATASET_FOLDER)
+Y_TEST_PATH = "{}y_test.npy".format(DATASET_FOLDER)
```
Comment (XanX3601) [inline utils/datasets.py:11]: A bit strange to have `folders.py` containing paths to directories and no `files.py` that would contain paths to interesting files but it's a detail

**Review by XanX3601 — CHANGES_REQUESTED** [review]
I would recommend some changes, please see my commentary. 
I made suggestion for `README` file but they lack ` ``` ` at the end because Github consider that they end the suggestion, please don't commit them directly.

Otherwise, it seems clean and good to go

File: requirements.txt (hunk 5 above)
Comment (ElieKadoche) [reply requirements.txt]: Corrected in https://github.com/XanX3601/deep_sentiment140/pull/3/commits/4833fdfb78d9b6d5d7314cc5c237879d8ad4c360.

File: download_dataset.py (hunk 4 above)
Comment (ElieKadoche) [reply download_dataset.py:25]: The file is small but this could be a hancement.

File: download_dataset.py (hunk 3 above)
Comment (ElieKadoche) [reply download_dataset.py]: Corrected in https://github.com/XanX3601/deep_sentiment140/pull/3/commits/4833fdfb78d9b6d5d7314cc5c237879d8ad4c360.

File: create_datasets.py (hunk 2 above)
Comment (ElieKadoche) [reply create_datasets.py]: Corrected in https://github.com/XanX3601/deep_sentiment140/pull/3/commits/4833fdfb78d9b6d5d7314cc5c237879d8ad4c360.

File: README.md (hunk 1 above)
Comment (ElieKadoche) [reply README.md]: Corrected in https://github.com/XanX3601/deep_sentiment140/pull/3/commits/4833fdfb78d9b6d5d7314cc5c237879d8ad4c360.

File: utils/datasets.py (hunk 7 above)
Comment (ElieKadoche) [reply utils/datasets.py:11]: Fair comment. I did that because we have general folders with no links between them. But concerning the files, they all share the same purpose: the dataset. So I found it more appropriate to put them in `datasets.py`.

**Review by XanX3601 — APPROVED** [review]
Good to go

**Comment by fmdinh** [comment]
Hi,

I have a question regarding the dataset. 160k is really heavy and it took several times for training. Do you think it will be better if we work for a small dataset, for example only 5k. After comparing our different models, we choose the best model to train in the 160k dataset?

**Comment by ElieKadoche** [comment]
You can do whatever you want for training the model. You can take all samples, you can take only one, you can randomly picked different ones after each epoch, etc. It is up to you.

Limiting the train dataset for all the models is a bad idea: you are not forced to use it all to train your model, but other methods could exploit the 160000 samples.

Note: you must not use the test dataset, it will be used to compare our models efficency.
